scalyr_agent/monitor_utils/k8s.py

```diff
@@ -1488,15 +1542,22 @@ def query_api( self, path, pretty=0, return_temp_errors=False, rate_limited=Fals
                 else:
                     raise K8sApiException( "Invalid response from Kubernetes API when querying '%s': %s" %( path, str( response ) ), status_code=response.status_code )
 
-            # Optionally append response text into logged response file
-            if logged_response_file:
-                logged_response_file.write(response.text)
+            # Optionally prepend stack trace into logged response
+            # newlines should become literal '\n' so that the entire response is a single line
+            if log_responses and response.text:
+                logged_response.append(response.text.replace('\n', '\\n'))
 
             return util.json_decode( response.text )
 
         finally:
-            if logged_response_file:
-                logged_response_file.close()
+            # Only debug-log the response if all criteria (status code, latency, response len) are satisfied
+            if log_responses:
+                if (response_status_code//100) != 2 or not self.log_api_exclude_200s:
```

Comment:
Could you add a comment explaining what this is doing.  This is the first time I'd ever seen `//`